src/Pact/Repl/Lib.hs

```diff
@@ -275,10 +281,27 @@ setsigs i [TList ts _ _] = do
   ks <- forM ts $ \t -> case t of
           (TLitString s) -> return s
           _ -> argsError i (V.toList ts)
-  setenv eeMsgSigs (S.fromList (map (PublicKey . encodeUtf8) (V.toList ks)))
+  setenv eeMsgSigs (V.fromList (map ((,mempty) . PublicKey . encodeUtf8) (V.toList ks)))
   return $ tStr "Setting transaction keys"
 setsigs i as = argsError i as
 
+setsigs' :: ZNativeFun LibState
+setsigs' i [TList ts _ _] = do
+  sigs <- forM ts $ \t -> case t of
+    (TObject (Object (ObjectMap om) _ _ _) _) -> do
+      case (M.lookup "key" om,M.lookup "clist" om) of
+        (Just (TLitString k),Just (TList clist _ _)) -> do
+          caps <- forM clist $ \cap -> case cap of
+            (TApp a _) -> view _1 <$> appToCap a
+            _ -> evalError' i $ "Expected capability invocation"
```

Comment:
might be good to relay which cap was not a cap in the failure message here

Reply:
`i` takes care of that